compiler/src/iree/compiler/Codegen/Common/GPU/BUILD.bazel

```diff
@@ -60,6 +60,7 @@ iree_compiler_cc_library(
         "GPUTileReduction.cpp",
         "GPUVectorization.cpp",
         "Passes.cpp",
+        "Utils.cpp",
```

Comment:
https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp exists for these kinds of shared patterns.

Reply:
IREE's requires the Bazel and CMake builds to match, but if you've updated the Bazel there is a script to automatically create the CMake changes
```
python build_tools/bazel_to_cmake/bazel_to_cmake.py
```
(https://github.com/openxla/iree/blob/main/docs/developers/developing_iree/testing_guide.md#configuring-the-build-system)

Reply:
> https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp exists for these kinds of shared patterns.

The GPUUtils file seems to be actual utilities for vectors or GPU patterns, while this utility file is for GPU Transforms. I would personally prefering keeping them here, as they make more sense for me for them to live here, But I do not have a strong opinion on this and will move them to Utils/GPUUtils.cpp/.h if you prefer that.

Reply:
Good point. I think we keep the implementation in WorkGroupSwizzle.cpp and add the entry function to [Passes.h](https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Codegen/Common/Passes.h). (Similar to the way GPUPipelining is handled [here](https://github.com/openxla/iree/blob/8fe8689cd78e9beaac568f9873778de9f751344a/compiler/src/iree/compiler/Codegen/Common/GPU/Passes.h#L45).